physlr/physlr.py

```diff
@@ -400,14 +400,49 @@ def diameter_of_tree(g, weight=None):
         return (u, v, diameter)
 
     @staticmethod
-    def determine_backbones_of_trees(g):
-        "Determine the backbones of the maximum spanning trees."
+    def detect_junctions_of_tree(gcomponent, messages, junction_threshold):
+        """"
+        detect the junctions in the tree,
+        with more than 2 branches larger than junction_threshold.
+        """
+        candidate_junctions = {node
+                               for node in list(gcomponent.nodes)
+                               if gcomponent.degree(node) > 2}
+        junctions = []
+        for candidate in candidate_junctions:
```

Comment:
```suggestion
        for junction in candidate_junctions:
```
I suggest `junction` or `candidate_junction` for this variable name.